crates/iceberg/src/scan.rs

```diff
@@ -218,6 +230,18 @@ impl TableScan {
 
                 let mut manifest_entries = iter(manifest.entries().iter().filter(|e| e.is_alive()));
                 while let Some(manifest_entry) = manifest_entries.next().await {
+
+                    if let Some(ref bound_predicate) = bound_predicate {
```

Comment:
move the check into `InclusiveMetricsEvaluator` (see other comment), which returns ROWS_MIGHT_MATCH if bound_predicate is None. This way we could entangle the already involved plan_files method?